tools/internal/coverage/main.go

```diff
@@ -39,15 +50,45 @@ func FindCoverageFiles(p string) {
 	check(err)
 }
 
+func ReadConfigData() *CodeCoverage {
+	jsonFile, err := os.Open(configFile)
+	check(err)
+	defer jsonFile.Close()
+
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	check(err)
+
+	var cov CodeCoverage
+	err = json.Unmarshal([]byte(byteValue), &cov)
+	check(err)
+	return &cov
+}
+
+// This supports doing a single package at a time. If this needs to be expanded in the future
+// this method will have to return a []*float64 for each packages goal
+func findCoverageGoal(covFiles []string, configData *CodeCoverage) *float64 {
+	var ret float64
+
+	for _, covFile := range covFiles {
+		for _, p := range configData.Packages {
+			if strings.Contains(covFile, p.Name) {
+				return &p.CoverageGoal
```

Comment:
Is there a reason to use a pointer to a float here and not just a float? It looks like you de-reference it later on anyway.

Reply:
When I was passing it in as a flag the parsed flag was a pointer to a float so probably for consistency, but I guess in hindsight I'll say I was extra concerned about memory management 😆

Reply:
Interestingly, I think the pointer and float64 end up being the same size in this case. So the pointer is actually less efficient (requires the pointer _and_ the float64!).

Reply:
I have been foiled again! @richardpark-msft

Reply:
I've found in some cases with the Azure SDKs (I forget why, perhaps because of auto-generated openAPI stuff and/or the flags as you say), there is a very specific reason that pointers needed to be used for basic types. But in all other cases, I think it's unnecessary.

Reply:
Also I wouldn't be as concerned about memory management/pre-mature optimization for tools code as opposed to customer-facing code.